rust/arrow/src/error.rs

```diff
@@ -90,6 +91,9 @@ impl From<serde_json::Error> for ArrowError {
 impl Display for ArrowError {
     fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
         match self {
+            ArrowError::Unsupported(source) => {
```

Comment:
What do you think of `Unimplemented` instead of `Unsupported`? Just to be consistent with the `unimplemented!` macro that rust already offers.

Reply:
I found a better name from reading parquet sources, It's called `NYI` (Not yet implemented), I think it's better to use similar names accross packages.

Thanks